File: java/pre15/src/org/garret/perst/AssertionFailed.java
package org.garret.perst;

/**
 * Exception raised by <code>Assert</code> class on failed assertion.
 */
public class AssertionFailed extends Error {
    AssertionFailed() { 
        super("Assertion failed");
    }

    AssertionFailed(String description) { 
        super("Assertion '" + description + "' failed");
    }
}

File: csharp/src/AssertionFailed.cs
namespace TenderBase
{
    using System;
    
    /// <summary> Exception raised by <code>Assert</code> class when assertion was failed.</summary>
    [Serializable]
    public class AssertionFailed : System.ApplicationException
    {
        internal AssertionFailed()
            : base("Assertion failed")
        {
        }

        internal AssertionFailed(string description)
            : base("Assertion '" + description + "' failed")
        {
        }
    }
}
